### ISSUE TITLE
Getting started guide fails at init step

### ISSUE THREAD

**Issue by michaellashinsky** [Microsoft/react-native-windows#1638]
### Environment

1. `react-native -v`:
react-native-cli: 2.0.1
react-native: 0.53.0
2. `npm ls rnpm-plugin-windows`:
rnpm-plugin-windows@0.2.8
3. `npm ls react-native-windows`:
react-native-windows@0.52.0-rc.2
4. `node -v`:
v9.5.0
5. `npm -v`:
5.6.0
6. `yarn --version`:
1.3.2

- Target Platform: Can't even init let alone build

- Target Platform Version(s): None yet

- Target Device(s): Desktop eventually

- Development Operating System: Windows 10 Enterprise Build 16299

- Visual Studio Version: Visual Studio 2017 Community Version 15.5.6

### Steps to Reproduce

1.  Goto https://github.com/Microsoft/react-native-windows/blob/master/docs/GettingStarted.md
2.  Follow the steps to get setup using powershell (sans choco install)
3.  Get to step of issuing command `react-native windows` and observe error "windows not recognized"
4.  Run `npm i` to ensure all the dependencies get installed (it appears they don't with the init) and observe 729 added packages and 37 updated packages
5.  Issue `react-native windows` again and observe "Could not find react-native-windows@0.53.*." message
6.  Issue `npm i --save react-native-windows` command and observe packages added to package.json and react-native-windows is installed at 0.52.0-rc.2 (expected 0.53.0) also added 305 packages.
7.  Issue `react-native windows` again and again observe error in step 5.
8.  Issue `npm i` again just to be sure all dependencies are correct and installed.  Observe warning react-native-windows@0.52.0-rc.2 requires peer of react-native@~0.52.0 but none is installed you must install peer dependencies yourself.
9.  Issue `npm i --save react-native@0.52.0` (at this point getting aggravated) and observe that react-native@0.52.0 is installed
10. Issue `react-native windows` again and observe "Could not find react-native-windows@0.52.*."
11.  Delete node_modules directory and issue `npm i` yet again (last ditch effort) and observe 1165 packages installed along with a few optional dependencies not being installed
12. Issue `react-native windows` again and again "Could not find react-native-windows@0.52.*." message.

### Expected Behavior

I thought I'd at a minimum follow the Getting started guide and have a hello world app.

### Actual Behavior

See steps for details on what went wrong

### Reproducible Demo

https://github.com/Microsoft/react-native-windows/blob/master/docs/GettingStarted.md

**Comment by rozele** [comment]
Thanks for reporting @michaellashinsky. Can you debug the `react-native windows` command by setting a breakpoint at https://github.com/Microsoft/react-native-windows/blob/aae6490751034d31e3b369a5d0115db3ba3dcff2/local-cli/rnpm/windows/src/common.js#L50? Do you have a custom NPM registry configured? What's the output of:
```
npm config get registry
```

**Comment by michaellashinsky** [comment]
Registry is standard as the output is https://registry.npmjs.org/

I can try the debug steps later this afternoon, go a few meetings today and some work to complete.

**Comment by michaellashinsky** [comment]
@rozele I think I found the issue and we can most likely close this bug.  It looks like my company has a proxy server that was trying to resolve the yeoman calls to pull down dependencies.  I configured my npmrc to point to the proxy and it explains why I can call the registry and get my packages.  In order for this to work at work I'd also need to configure an rc file for yeoman-generator so that it too can find its way to the repo.  Jumping on the guest-wifi (shhhh don't tell) and removing the proxy settings in my npmrc file I can at least appear to get moving passed the original issue.  I moved back to 0.50.0 of react-native and react-native-windows as another post suggested and here's my PS output, The debug module is something I can dig into later, as I said I am pretty busy till later today.

```PS C:\sandbox\personal\myReactNativeProject> react-native windows
Scanning folders for symlinks in C:\sandbox\personal\myReactNativeProject\node_modules (48ms)
Reading application name from package.json...
Reading react-native version from node_modules...
Checking for react-native-windows version matching 0.50.*...
Installing react-native-windows@0.50.0...
warning react-native-windows > yeoman-generator > download > gulp-decompress > gulp-util@3.0.8: gulp-util is deprecated
- replace it, following the guidelines at https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5
warning " > react-native@0.50.4" has incorrect peer dependency "react@16.0.0".
warning " > babel-jest@22.2.2" has unmet peer dependency "babel-core@^6.0.0 || ^7.0.0-0".
warning " > react-native-windows@0.50.0" has incorrect peer dependency "react@16.0.0".
react-native-windows@0.50.0 successfully installed.
Cannot find module 'debug'

**Comment by michaellashinsky** [comment]
@rozele I can confirm that this was the problem for the setup.  That being said this issue can be closed I did face other challenges.  
- I still needed to `npm i` after `npm i --save rnpm-plugin-windows` as for whatever reason the windows command was not recognized. Betting on my yarn installation causing this as running `yarn` didn't fix this issue.  It was for sure fixed after `npm i`.
 - After the aforementioned work-around I could then call `react-native windows` and I observed an error saying debug could not be found.  I checked for a windows directory and it was missing.   I ran `react-native windows` again and the issue was fixed. Not sure why I don't think I have anything else weird in my env.
- I then opened VS2017 Community Edition and continued down the instructions.  When I got to the step that requires Debug -> Start without Debugging I got the 403 red box issue.  I dug around some on the web and found that react-native wants to start the project on 8081, problem is my McAfee does too.  - I started the `react-native start` command with a new --port 8090 and re-ran the project running into the issue again but noticed the app had the old port.  I then dug into the project node libs and changed the ReactNative.Shared\DevSupport\DevServerHelper.cs line 22 to point to this new 8090 port and restarted the app.  All "appears" to be working now.  Apologies for the proxy issue, but this other stuff might be best served by updating the docs.
